packages/adapter-api/src/change.ts

```diff
@@ -92,6 +92,13 @@ export type DetailedChange<T = ChangeDataType | Values | Value> =
     path?: ReadonlyArray<string>
   }
 
+export type AuditInformation = {
+  changedBy?: string
+  changedAt?: string
+  createdBy?: string
+  createdAt?: string
+ }
```

Comment:
Seems like `audit` is always defined in `FetchChange`.
I think that it would be better to leave it with the `audit?` as you did, but modify `getAuditInformationFromElement` to return `AuditInformation | undefined` and in addition guard the `AuditInformation` type definition so it allow be an empty object:

```suggestion
export type AuditInformation = {
  changedBy?: string
  changedAt?: string
  createdBy?: string
  createdAt?: string
} & ({ changedBy: string } | { changedAt: string } | { createdBy: string } | { createdAt: string })
```

WDYT?